src/doc/rustdoc/src/write-documentation/documentation-tests.md

```diff
@@ -376,6 +376,60 @@ that the code sample should be compiled using the respective edition of Rust.
 # fn foo() {}
 ```
 
+Starting in the 2024 edition[^edition-note], compatible doctests will be merged as one before being
+run. It means that they will share the process, so any change to global/static variables will now
+impact the other doctests. We combine doctests for performance reasons: the slowest part of
+doctests is to compile them. Merging all of them into one file and compiling this new file, then
```

Comment:
Is compiling really the slowest part?

> core	4604	54.08s	23.20s

That's
* 54.08s/4604tests = 11ms/test on master
* 23.2/4604tests = 5ms/test with this PR

For comparison, on master and my machine I get 
18.84s/4709tests = 4ms per doctest
1s/1673tests = 0.6ms per libtest

So there's still a big gap to libtests. Is it using a different test harness compared to lib tests?

Reply:
Perhaps part of the problem is that some tests still have to be compiled/run separately -- e.g. `compile_fail` -- and that's making the average look worse? But I don't know if that's all of it. I do wonder if #63827 was the right change to make given the cost of spawning processes.

Reply:
Another factor (though again *probably* not the most significant factor) is that rustdoc essentially parses doctests twice: once to check their syntax and once when it runs `rustc` on that doctest or the merged doctest that contains it.

Reply:
I added information in the documentation that was provided in the PR comments.